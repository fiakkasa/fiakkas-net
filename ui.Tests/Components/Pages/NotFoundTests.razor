@inherits TestContext

@code {

    [Fact]
    public void NotFound_Renders_Template()
    {
        var cut = Render(@<NotFound/>);

        cut.FindComponent<UniversalTitle>();
        var headContent = cut.FindComponent<HeadContent>();
        var expectedHeadRenderedFragment = Render(headContent.Instance.ChildContent!);

        var title = cut.FindAll("h1.page-header");

        Assert.Empty(title);
        Assert.NotNull(headContent.Instance.ChildContent);
        expectedHeadRenderedFragment.MarkupMatches(@<meta name="robots" content="noindex"/>);
        cut.Markup.MatchSnapshot();
    }

}
