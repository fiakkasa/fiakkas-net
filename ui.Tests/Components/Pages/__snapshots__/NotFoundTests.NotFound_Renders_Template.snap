



<div class="d-flex flex-column align-items-center gap-2"><img src="./images/404-background.png" class="img-fluid rounded-5" alt="404">
    <div class="display-1">4🙃4</div>
    <div class="lead">Whoops, the page you are looking for seems to be unavailable...</div>
    <a href="./" class="btn btn-lg btn-outline-primary">Home</a></div>
